test-apps/display-test-app/src/frontend/App.ts

```diff
@@ -75,6 +75,45 @@ class SVTSelectionTool extends SelectionTool {
   }
 }
 
+class SignInTool extends Tool {
+  public static override toolId = "SignIn";
+  public override run(): boolean {
+    signIn(); // eslint-disable-line @typescript-eslint/no-floating-promises
+    return true;
+  }
+}
+
+abstract class PushPullChangesTool extends Tool {
+  public static override get maxArgs() { return 1; }
+
+  protected abstract execute(bc: BriefcaseConnection, arg?: string): Promise<void>;
+
+  public override run(arg?: string): boolean {
+    const imodel = IModelApp.viewManager.selectedView?.iModel;
+    if (!imodel || !imodel.isBriefcaseConnection())
+      return false;
+
+    this.execute(imodel, arg); // eslint-disable-line @typescript-eslint/no-floating-promises
+    return true;
+  }
+}
+
+class PushChangesTool extends PushPullChangesTool {
+  public static override toolId = "PushChanges";
+
+  protected override async execute(bc: BriefcaseConnection, description?: string): Promise<void> {

```

Comment:
I'd make description required. It sets a bad precedent to allow meaningless change descriptions.